packages/transport/src/modules/relationshipTemplates/local/RelationshipTemplate.ts

```diff
@@ -50,7 +55,16 @@ export class RelationshipTemplate extends CoreSynchronizable implements IRelatio
     }
 
     public toRelationshipTemplateReference(): RelationshipTemplateReference {
-        return RelationshipTemplateReference.from({ id: this.id, key: this.secretKey, forIdentityTruncated: this.cache!.forIdentity?.toString().slice(-4) });
+        return RelationshipTemplateReference.from({
+            id: this.id,
+            key: this.secretKey,
+            forIdentityTruncated: this.cache!.forIdentity?.toString().slice(-4),
+            passwordType: this.passwordType
+        });
+    }
+
+    public get passwordType(): number | undefined {
```

Comment:
In principle, it is good to have a distinction between PINs and "regular" passwords, especially when it comes to their representation in the UI. However, I find that the chosen approach can be problematic if my specified `password` is used to decide for me whether it is a PIN or not and what the UI representation is. Implicitly deriving the password type from my chosen password gives the App user the opportunity to draw conclusions about my password via the UI, without me necessarily having been aware of this.

I would find it more transparent and secure for Integrators of Connectors to have two optional fields `password` and `pin`, which both have an individual validation and representation in the UI.

Even if we don't need to tackle this in this PR, as it would change a lot, I just wanted to share my impression.